crates/symbolicator/src/services/symbolication.rs

```diff
@@ -1894,125 +2020,102 @@ impl SymbolicationActor {
-                            let return_address =
-                                frame.return_address(minidump_state.system_info.cpu_arch);
-
-                            frames.push(RawFrame {
-                                instruction_addr: HexValue(return_address),
-                                package: frame.module().map(CodeModule::code_file),
-                                trust: frame.trust(),
-                                ..RawFrame::default()
-                            });
-                        }
-
-                        stacktraces.push(RawStacktrace {
-                            is_requesting: requesting_thread_index.map(|r| r == index),
-                            thread_id: Some(thread.thread_id().into()),
-                            registers,
-                            frames,
-                        });
-                    }
-
-                    Ok(procspawn::serde::Json(StackWalkMinidumpResult {
-                        modules,
-                        missing_modules,
-                        stacktraces,
-                        minidump_state,
-                    }))
+                |(cfi_caches, minidump_path, spawn_time)| {
+                    stackwalk_with_breakpad(cfi_caches, minidump_path, spawn_time)
                 },
             );
 
-            Self::join_procspawn(
+            let result_old = Self::join_procspawn(
                 spawn_result,
                 Duration::from_secs(60),
                 "minidump.stackwalk.spawn.error",
-            )
+            )?;
+
+            let result_new = if compare_stackwalking_methods {
+                let spawn_time = std::time::SystemTime::now();
+                let spawn_result = pool.spawn(
+                    (
+                        procspawn::serde::Json(cfi_caches),
+                        minidump_path,
+                        spawn_time,
+                    ),
+                    |(cfi_caches, minidump_path, spawn_time)| {
+                        stackwalk_with_rust_minidump(cfi_caches, minidump_path, spawn_time)
+                    },
+                );
+
+                match Self::join_procspawn(
+                    spawn_result,
+                    Duration::from_secs(60),
+                    "minidump.stackwalk_new.spawn.error",
+                ) {
+                    Ok(result) => Some(result),
+                    Err(e) => {
+                        sentry::capture_error::<dyn std::error::Error>(e.as_ref());
+                        None
+                    }
+                }
+            } else {
+                None
+            };
+
+            // Determine if there was a stackwalking difference or the new method performed poorly.
+            // If so, the minidump will be saved further down after some more processing.
+            let problem = result_new.and_then(|result_new| {
+                    metric!(timer("minidump.stackwalk.duration") = result_new.duration, "method" => "new");
```

Comment:
this is nitpicking, but can we be a little more descriptive and use "rust-minidump" and "breakpad" instead of "new" and "old" in the metric tag names?